addon/components/select-2.js

```diff
@@ -463,6 +498,22 @@ var Select2Component = Ember.Component.extend({
         // treat data as a single object
         value = get(data, optionValuePath);
       }
+    } else if (this.get('tags') && e) {
+      if (e.added) {
+        var record;
+        if (e.added instanceof DS.Model) {
+          record = e.added;
+        } else {
```

Comment:
I need to check the type of the `value` property here, so it can both support ember data record arrays and simple objects.